src/btree/bt_curprev.c

```diff
@@ -664,6 +664,10 @@ __wt_btcur_prev(WT_CURSOR_BTREE *cbt, bool truncating)
 
     WT_STAT_CONN_DATA_INCR(session, cursor_prev);
 
+    /* Reposition the cursor if the cursor was reset internally. */
+    if (F_ISSET(cbt, WT_CBT_REPOSITION) && session->txn->isolation == WT_ISO_SNAPSHOT)
```

Comment:
I think the second condition "session->txn->isolation .." is not required, as we don't release page if it not snapshot isolation.  
I would like to restrict the release page only for snapshot isolation, and if the page is not released the flag "WT_CBT_REPOSITION" will not be set.